### PR TITLE
Remove rollout options for Rust Enhancers

### PR DESCRIPTION
**Pull request by Swatinem**
We have fully enabled these, so lets remove all the options usage related to it.

This is pretty much a followup from #65533.

### CODE REVIEW

File: src/sentry/grouping/enhancer/__init__.py [hunk 1]
```diff
@@ -299,13 +279,15 @@ def apply_modifications_to_frame(
             self.rust_enhancements, match_frames, exception_data
         )
 
-        if rust_enhanced_frames and prefer_rust_enhancers():
```
Comment (armenzg) [inline src/sentry/grouping/enhancer/__init__.py:302]: Do we need to remove `prefer_rust_enhancers` now that it is not being used?

File: src/sentry/grouping/enhancer/__init__.py [hunk 2]
```diff
@@ -19,7 +18,7 @@
 from sentry_ophio.enhancers import Cache as RustCache
 from sentry_ophio.enhancers import Enhancements as RustEnhancements
 
-from sentry import options, projectoptions
+from sentry import projectoptions
 from sentry.features.rollout import in_random_rollout
```
Comment (armenzg) [inline src/sentry/grouping/enhancer/__init__.py:22]: If we remove `prefer_rust_enhancers` we can remove this import.

**Review by armenzg — COMMENTED** [review]
Excited to see this!

Do we need to remove any tests that got added which would not make any more sense after removing this?

File: src/sentry/grouping/enhancer/__init__.py (hunk 2 above)
Comment (Swatinem) [reply src/sentry/grouping/enhancer/__init__.py:22]: Indeed, I completely missed this.

**Comment by Swatinem** [comment]
> Do we need to remove any tests that got added which would not make any more sense after removing this?

I removed a bunch of unused test decorators now.

Now we have the problem that some tests are testing broken enhancement rules and those are logging errors to the console but then falling back to the python implementation.

That, plus support for `config_structure version=2`, which we did not implement.

Tests are passing with this, but they are spamming stuff to the console. Wouldn’t be the first time though that tests are spamming the console (looking at you, code mappings)

**Review by armenzg — APPROVED** [review]

**Pull request merged**